tests/bunni/test_flow.py

```diff
@@ -0,0 +1,73 @@
+from brownie import interface, chain
+import pytest
+
+
+@pytest.fixture(scope="function", autouse=True)
+def deposited(safe, bunni, badger, wbtc):
+    wbtc_amount = 1e8
+    badger_amount = 10_000e18
+
+    wbtc_before = wbtc.balanceOf(safe)
+    badger_before = badger.balanceOf(safe)
+
+    # https://github.com/ethereum/EIPs/issues/20#issuecomment-263524729
+    badger.approve(bunni.router, 0)
+    badger.approve(bunni.hub, 0)
+
+    assert badger.allowance(safe, bunni.router) == 0
+    assert badger.allowance(safe, bunni.hub) == 0
+
+    bunni_token_addr = bunni.deposit(wbtc_amount, badger_amount)
+
+    assert interface.IBunniToken(bunni_token_addr).balanceOf(safe) > 0
+    assert wbtc.balanceOf(safe) < wbtc_before
+    assert badger.balanceOf(safe) < badger_before
+
+
+def test_withdraw(safe, bunni, badger, wbtc):
+    wbtc_before = wbtc.balanceOf(safe)
+    badger_before = badger.balanceOf(safe)
+
+    bunni_token_addr = bunni.hub.getBunniToken(bunni.bunni_key)
+
+    bunni_token = interface.IBunniToken(bunni_token_addr)
+    bunni.withdraw()
+
+    assert bunni_token.balanceOf(safe) == 0
+    assert wbtc.balanceOf(safe) > wbtc_before
+    assert badger.balanceOf(safe) > badger_before
+    chain.reset()
+
+
+def test_stake_and_unstake(safe, bunni):
+    bunni_token_addr = bunni.hub.getBunniToken(bunni.bunni_key)
+    bunni_token = interface.IBunniToken(bunni_token_addr)
+
+    gauge_addr = bunni.deploy_gauge(1e18)
+    gauge = safe.contract(gauge_addr)
+
+    before_bunni = bunni_token.balanceOf(safe)
+    before_gauge = gauge.balanceOf(safe)
+
+    bunni.stake(gauge_addr)
+
+    assert bunni_token.balanceOf(safe) < before_bunni
+    assert gauge.balanceOf(safe) > before_gauge
+
+    before_bunni = bunni_token.balanceOf(safe)
+    before_gauge = gauge.balanceOf(safe)
+
+    bunni.unstake(gauge_addr)
+
+    assert gauge.balanceOf(safe) < before_gauge
+    assert bunni_token.balanceOf(safe) > before_bunni
+
+
+def test_compound(bunni, badger, wbtc):
+    badger.approve(bunni.router, 0)
+
+    # simulate some trading so we have some fees to compound
+    bunni.swap([badger, wbtc], 1000e18)
```

Comment:
the intention here i think is to use another class (univ3), it reverts locally

Reply:
nvm i see the inheritance, does it revert at your end?

Reply:
works on my end, all tests passing currently

Reply:
not blocking, since its not inherited from bunni class itself, but locally for me was reverting

<img width="628" alt="Screenshot 2023-03-15 at 01 05 17" src="https://user-images.githubusercontent.com/84875062/225117428-c2a27f09-9238-46d8-bbde-0cda95edace7.png">

<img width="609" alt="Screenshot 2023-03-15 at 01 06 57" src="https://user-images.githubusercontent.com/84875062/225117667-f36d5e3b-cdc8-4e5c-88f6-d98c7bc1f790.png">

Reply:
reset my config and delete `deployments.db` still the revert persist somehow

tried with `min_out=0` for the wbtc -> badger swap. can confirm that the test passes successfully ✅

Reply:
im getting the same. somehow the second `swap` call is causing this. if however, i only make one swap, it reverts on `assert added_liq > 0`; `self.hub.compound(self.bunni_key).return_value` comes back as `(0, 0, 0)`

Reply:
very strange. obviously these swaps will happen naturally on mainnet, so not breaking. maybe we need some univ3 tests